monsterCatalog.py

```
@@ -0,0 +1,280 @@
+@app.route('/signin')
+def sign_in():
+    state = ''.join(random.choice(string.ascii_uppercase + string.digits)
+                    for x in xrange(32))
+    flask.session['state'] = state
+    return flask.render_template('signup.html', STATE=state)
+
+
+@app.route('/create', methods=['GET', 'POST'])
+def create_monster():
+    # print flask.request.form
+    if 'username' not in flask.session:
+        return flask.redirect('/signin')
+    if flask.request.method == 'POST':
+        if 'encoded_picture' in flask.request.form:
+            encoded_picture = flask.request.form.get('encoded_picture')
+            picture = base64.b64decode(str(encoded_picture))
+            picture_id = str(uuid.uuid4())
+            picture_url = 'https://s3.amazonaws.com/monster-catalog/' + picture_id
+
+            session = boto3.session.Session(
+                aws_access_key_id=app.config['AWS_ACCESS_KEY_ID'],
+                aws_secret_access_key=app.config['AWS_SECRET_ACCESS_KEY']
+            )
+            s3 = session.resource(service_name='s3')
+
+            s3.Bucket(app.config['AWS_S3_BUCKET']).put_object(
+                Key=picture_id,
+                Body=picture,
+                ContentType='image/png')
+
+            monster = Monster(
+                name=flask.request.form['name'],
+                diet=flask.request.form['diet'],
+                enjoys=flask.request.form['enjoys'],
+                creator=flask.session['user_id'],
+                picture=picture_url,
+                intentions=flask.request.form['intentions'],
+                created_date=datetime.datetime.now()
+                )
+            db_session.add(monster)
+            try:
+                db_session.commit()
+                print "committed"
+            except Exception as e:
+                db_session.rollback()
+                print "failed"
+                print e
+            return flask.url_for('main_page')
+        else:
+            pass
```

This should never happen, right?  You might want to raise an error or redirect in this case.